nextstrain/cli/remote/nextstrain_dot_org.py

```diff
@@ -600,6 +633,37 @@ def raise_for_status(response: requests.Response) -> None:
             raise
 
 
+def authn_challenge(response: requests.Response) -> Optional[Dict[str, str]]:
+    """
+    Extract the Bearer authentication challenge parameters from the HTTP
+    ``WWW-Authenticate`` header of *response*.
+
+    >>> r = requests.Response()
+    >>> r.headers["WWW-Authenticate"] = 'Bearer error="invalid_token", error_description="token is expired"'
+    >>> authn_challenge(r)
+    {'error': 'invalid_token', 'error_description': 'token is expired'}
+
+    >>> r = requests.Response()
+    >>> r.headers["WWW-Authenticate"] = 'Basic realm="nunya"'
+    >>> authn_challenge(r) # None
+
+    >>> r = requests.Response()
+    >>> authn_challenge(r) # None
+    """
+    challenge = response.headers.get("WWW-Authenticate")
+
+    if not challenge:
+        return None
+
+    challenge_params = remove_prefix("Bearer ", challenge, ignore_case = True)
```

Comment:
_[non-blocking]_

Is there a particular reason to ignore case here? I don't know much about OAuth but all the examples I'm seeing use the `Bearer` casing.

Reply:
Postel's Law… but maybe unnecessarily so. The RFC [mandates `Bearer` literally](https://datatracker.ietf.org/doc/html/rfc6750#section-2.1). I also note when checking related RFCs that challenges (what appear in `WWW-Authenticate` and elsewhere) allow multiple spaces (`1*SP`) between the `auth-scheme` (e.g. `Bearer`) and the params, whereas the code in this branch trims only 1 space and the rest will end up in the name of the first parameter (likely breaking parsing if present).

I'll fix these two things up.

Reply:
Done: https://github.com/nextstrain/cli/compare/949971d3cec6a9334893ddc0e49216c836b3ad16..6a62b60fba7427ffee8f277d0002303928b7b15b